File: backend/app/schemas/laundry_trust_score.py
"""Laundry trust score API schemas."""

from __future__ import annotations

from datetime import datetime
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel, Field

from app.models.enums import LaundryTrustLevel

LAUNDRY_TRUST_LEVEL_LABELS: dict[LaundryTrustLevel, str] = {
    LaundryTrustLevel.premium: "Premium",
    LaundryTrustLevel.trusted: "Trusted",
    LaundryTrustLevel.verified: "Verified",
    LaundryTrustLevel.under_review: "Under Review",
}


class LaundryTrustMetrics(BaseModel):
    on_time_delivery_pct: float = Field(description="0-100 percentage")
    complaint_rate_pct: float = Field(description="Complaints per 100 completed orders")
    refund_rate_pct: float = Field(description="Refunds per 100 completed orders")
    dispute_rate_pct: float = Field(description="Active disputes per 100 completed orders")
    avg_rating: float
    review_count: int
    completed_orders: int


class LaundryTrustScoreSummary(BaseModel):
    laundry_id: UUID
    laundry_name: str
    city: str
    owner_name: str | None = None
    trust_score: int
    level: LaundryTrustLevel
    level_label: str
    metrics: LaundryTrustMetrics
    calculated_at: datetime


class LaundryTrustScoreDetail(LaundryTrustScoreSummary):
    score_breakdown: dict[str, float]

File: frontend/services/laundry-trust-score.ts
import { api, type ApiEnvelope } from '@/lib/api';

export type LaundryTrustLevel = 'premium' | 'trusted' | 'verified' | 'under_review';

export const LAUNDRY_TRUST_LEVEL_LABELS: Record<LaundryTrustLevel, string> = {
  premium: 'Premium',
  trusted: 'Trusted',
  verified: 'Verified',
  under_review: 'Under Review',
};

export interface LaundryTrustMetrics {
  on_time_delivery_pct: number;
  complaint_rate_pct: number;
  refund_rate_pct: number;
  dispute_rate_pct: number;
  avg_rating: number;
  review_count: number;
  completed_orders: number;
}

export interface LaundryTrustScoreSummary {
  laundry_id: string;
  laundry_name: string;
  city: string;
  owner_name: string | null;
  trust_score: number;
  level: LaundryTrustLevel;
  level_label: string;
  metrics: LaundryTrustMetrics;
  calculated_at: string;
}

export interface LaundryTrustScoreDetail extends LaundryTrustScoreSummary {
  score_breakdown: Record<string, number>;
}

export async function getPartnerTrustScore(): Promise<LaundryTrustScoreSummary> {
  const { data } = await api.get<ApiEnvelope<LaundryTrustScoreSummary>>('/partner/trust-score');
  return data.data;
}

export async function listAdminLaundryTrustScores(): Promise<LaundryTrustScoreSummary[]> {
  const { data } = await api.get<ApiEnvelope<LaundryTrustScoreSummary[]>>('/admin/laundry-trust-scores');
  return data.data;
}

export async function getAdminLaundryTrustScoreDetail(
  laundryId: string,
): Promise<LaundryTrustScoreDetail> {
  const { data } = await api.get<ApiEnvelope<LaundryTrustScoreDetail>>(
    `/admin/laundry-trust-scores/${laundryId}`,
  );
  return data.data;
}
